geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TiledRasterRDD.scala

```diff
@@ -316,6 +322,36 @@ class SpatialTiledRasterRDD(
   val rdd: RDD[(SpatialKey, MultibandTile)] with Metadata[TileLayerMetadata[SpatialKey]]
 ) extends TiledRasterRDD[SpatialKey] {
 
+  def resample_to_power_of_two(
+    col_power: Int,
+    row_power: Int,
+    resampleMethod: String
+  ): TiledRasterRDD[SpatialKey] = {
+    val method: ResampleMethod = TileRDD.getResampleMethod(resampleMethod)
+    val cols = 1<<col_power
+    val rows = 1<<row_power
+    val rdd2 = rdd.mapValues({ tile => tile.resample(cols, rows, method) })
+
+    val metadata = rdd.metadata
+    val metadata2 = TileLayerMetadata(
```

Comment:
What do you think about using `copy` to create a new instance of `TileLayerMetadata`?

```scala

val metadata = rdd.metadata
val metadata2 = metadata.copy(layoutDefinition = LayoutDefinition(metadata.extent, TileLayout(metadata.layout.tileLayout.layoutCols, metadata.layout.tileLayout.layoutRows, cols, rows))